jasm-cli/src/main/java/me/darknet/assembler/cli/commands/CompileCommand.java

```diff
@@ -163,8 +163,16 @@ public void run() {
                 ClassRepresentation representation = result.representation();
                 switch (MainCommand.target) {
                     case JVM -> {
+		        Path outputPath;
+                        if (output != null) {
+                           outputPath = output.toPath();
+                        } else {
+                           String inputFilename = source.isPresent() ? source.get().getName() : "output";
+                           String classFilename = inputFilename.endsWith(".java") ? inputFilename.substring(0, inputFilename.length() - 5) + ".class" : inputFilename + ".class";
```

Comment:
Why are we checking for `.java`? `.java` is not a valid input format for the decompile command

Reply:
im dumb, is .jasm the default format?

Reply:
It is, i never declared it anywhere. I assumed it would be self explanatory, but i guess not

Reply:
Alright